lib/frameworks/cucumber.js

```diff
@@ -1,14 +1,15 @@
-var ConfigParser = require('../configParser'),
-  q = require('q');
+var q = require('q');
 
 /**
  * Execute the Runner's test cases through Cucumber.
  *
  * @param {Runner} runner The current Protractor Runner.
  * @param {Array} specs Array of Directory Path Strings.
+ * @param {function(string): string} resolvePath resolves a path relative to the
+ *     config file
  * @return {q.Promise} Promise resolved with the test results
  */
-exports.run = function(runner, specs) {
+exports.run = function(runner, specs, resolvePath) {
```

Comment:
This seems like a messy way of passing this information.

The problem here is that some plugins or frameworks may want to add their own things to the configuration file which will need to be interpreted as file paths, which need to be resolved relative to something. This is fairly simple logic which the frameworks could probably do themselves, but we have an internal helper. Is it even worth exporting this internal helper? Most of the reason that `ConfigParser.resoveFilePatterns` is so long is ONLY to support cucumber stuff. 

I would say let's not export this function anywhere, simplify the internal implementation, and make cucumber do its own complicated logic with line numbers. What do you think?